treeherder/webapp/api/performance_serializers.py

```diff
@@ -142,6 +142,10 @@ class PerformanceAlertSummarySerializer(serializers.ModelSerializer):
     prev_push_revision = serializers.SlugRelatedField(read_only=True,
                                                       slug_field='revision',
                                                       source='prev_push')
+    assignee = serializers.SlugRelatedField(slug_field="username", allow_null=True,
```

Comment:
Does this need a default value?

Reply:
Is this supposed to be the id? If so, can you name it so it's `assignee_id`?

Reply:
> I think for this and assignee_email you should add `read_only=True`.

I need to be able to update this from frontend side. In the following related PRs, I want to add a 'Take' button, at alert summary level. It's a button very similar to that from Bugzilla.

When a user clicks it, the frontend will issue PUT request on the correct REST API, providing only the new assignee.

Reply:
> I think it'd be clearer to name it `assignee_username`.

Not really sure if/how I can do that. From what I can see, Django requires the same identifier for `PerformanceAlert.assignee` & `PerformanceAlertSerializer.assignee` fields.

Reply:
You can return `assignee__username` in the body of the API response and then rename it in the serializer: `assignee_username = serializers.CharField(source="username__assignee")`

Reply:
> > I think for this and assignee_email you should add `read_only=True`.
> 
> Well, I need to be able to update this from frontend side. In the following related PRs, I want to add a 'Take' button, at alert summary level. It's a button very similar to that from Bugzilla.
> 
> When a user clicks it, the frontend will issue PUT request on the correct REST API, providing only him as the new assignee.

Right, I'm just wondering if there's a way to prevent these fields from being accidentally overwritten during PUT operations. I think `editable=false` also makes it read-only. Perhaps there's a difference between updating with an entire user object versus one of the fields, or even using that assignee pk?

I think it's worth having @camd take a look at this pr before merging in case I'm missing something.

Reply:
> You can return `assignee__username` in the body of the API response and then rename it in the serializer: `assignee_username = serializers.CharField(source="username__assignee")`

Oh, I think I got it. Thanks for this trick!

Reply:
> > > I think for this and assignee_email you should add `read_only=True`.
> > 
> > 
> > Well, I need to be able to update this from frontend side. In the following related PRs, I want to add a 'Take' button, at alert summary level. It's a button very similar to that from Bugzilla.
> > When a user clicks it, the frontend will issue PUT request on the correct REST API, providing only him as the new assignee.
> 
> Right, I'm just wondering if there's a way to prevent these fields from being accidentally overwritten during PUT operations. I think `editable=false` also makes it read-only. Perhaps there's a difference between updating with an entire user object versus one of the fields, or even using that assignee pk?
> 
> I think it's worth having @camd take a look at this pr before merging in case I'm missing something.

@camd could you take a look over this PR? Are there risks during the PUT operations?